### PR TITLE
Add support for PSID format binding table

### PR DESCRIPTION
**Pull request by dpino**
Binding table using PSID format for port range is converted to our
internal binding table format.

Each entry in a PSID format binding table has four parameters for PSID:

{psid, k, m , a}

where:
- psid: port set identificator.
- k: psid length (psid <= 2^k) bits.
- m: maximum number of continue ports bits
- a: port restriction bits.

### CODE REVIEW

**Comment by dpino** [comment]
The PSID format binding table is equivalent to:

```
{
 {'127:2:3:4:5:6:7:128', '178.79.150.233', 1, 128, '8:9:a:b:c:d:e:f'},
 {'127:11:12:13:14:15:16:128', '178.79.150.233', 128, 58495},
 {'127:22:33:44:55:66:77:128', '178.79.150.15', 4, 8195},
 {'127:24:35:46:57:68:79:128', '178.79.150.2', 7808, 7935, '1E:1:1:1:1:1:1:af'},
 {'127:14:25:36:47:58:69:128', '178.79.150.3', 4096, 5119, '1E:2:2:2:2:2:2:af'}
}
```

**Comment by dpino** [comment]
This PR solves #101

File: src/apps/lwaftr/binding_table.lua [hunk 1]
```diff
@@ -33,15 +33,97 @@ local function pton_binding_table(bt)
...
-   if not bt_file then
-      error("bt_file must be specified or the BT pre-initialized")
+local function expand_row(range, row)
+   local v6, v4, br_v6 = row[1], row[2], row[4]
+   v6, v4 = ipv6:pton(v6), lwutil.rd32(ipv4:pton(v4))
+   if br_v6 then br_v6 = ipv6:pton(br_v6) end
+   return {v6, v4, range.start, range.last, br_v6}
+end
+
+local function coallesce(ranges)
```
Comment (aperezdc) [inline src/apps/lwaftr/binding_table.lua]: Typo: one “l” too much, it should be “[coalesce](http://www.wordreference.com/es/translation.asp?tranword=coalesce)”

**Comment by kbara** [comment]
The general concept of coalescing seems good. http://tools.ietf.org/html/rfc6346#section-5.2 mentions assigning multiple port ranges (potentially on multiple IPv4 IPs) to one customer, presumably (potentially) with one IPv6 address. Whether we coalesce or not seems like a pure implementation issue, but minimizing the size of the binding table by coalescing is nice. I think it'll slightly complicate dumping the table back out, but that should still be doable.

**Comment by kbara** [comment]
Specifying a, k, and m is redundant, because they add up to 16; you only need to specify 2 of the three.

**Comment by kbara** [comment]
This should also add the new test to .travis.yml.

**Comment by kbara** [comment]
I'd like to see a selftest() function in this, I think.

**Comment by kbara** [comment]
The example psid binding table can't be right. It has entries like:
``` {'127:24:35:46:57:68:79:128', '178.79.150.2', {29, 8, 7, 4}, '1E:1:1:1:1:1:1:af'},```
8 + 7 = 15, which leaves 1 bit for a, yet a is said to be 4 here.

**Comment by dpino** [comment]
@kbara: Updated PR with working `port_range` function. I tried to recreate the current binding table with the PSID format but there are tests that fail. Perhaps it's necessary to change the dest port in some packets. I commented the tests temporarily.

**Comment by kbara** [comment]
We need to update the docs to say what the {x, y, z} tuple is, too.

**Comment by kbara** [comment]
Let's do full a potentially != 0 PSID support. I fully expect it'll be awkward or worse to preserve our current port ranges with that. Let's change the sample PSID table to have one entry each, and change the port ranges in the tests accordingly (so the ones that should be bound all are, and the ones that shouldn't be all aren't - it might be worth testing those with off-by-one-from-bound bit patterns).

**Comment by dpino** [comment]
Superseded by #161.

**Pull request closed without merging**